miriway.cpp

```diff
@@ -203,6 +203,92 @@ struct ShellPids
         }
     }
 };
+
+class ShellComponentLaunchManager
+{
+public:
+    ShellComponentLaunchManager(ShellPids& shell_pids, ExternalClientLauncher& client_launcher)
+        : shell_launch([&](std::vector<std::string> const& cmd)
+        {
+            shell_pids.insert(client_launcher.launch(cmd), [this, cmd] {
+                relaunch(cmd);
+            });
+        })
+    {}
+
+    void relaunch(std::vector<std::string> const& cmd)
+    {
+        auto const now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
```

Comment:
Why are we converting to `time_t` here? Leaving everything as `std::chrono::time_point` and `std::chrono::duration` seems like it would be easier (and also let you remove `SECONDS` from the variable names, as the units will be in the type).